### PR TITLE
feat: Respect original parameter overrides with git write-back

### PR DESCRIPTION
**Pull request by ks-yim**
Modification:

Make image-updater respect the original parameter overrides written in `.argocd-source-<appName>.yaml`.

Motivation:

Currently, the original overridden parameters stored in `.argocd-source-<appName>.yaml`
are not respected when image-updater writes back its change to git.

Imagine that a user has originally defined `.argocd-source-<appName>.yaml` like below
```yaml
helm:
  parameters:
  - name: some.prop
    value: some-value
    forcestring: true
```
...and image-updater wants to write back the additional overrides below
- `foo.image.repository: foo`
- `foo.image.tag: 2.0.0`

The user may want the edited file to be like...
```yaml
helm:
  parameters:
  - name: some.prop
    value: some-value
    focestring: true
  - name: foo.image.repository
    value: foo
    forcestring: true
  - name: foo.image.tag
    value: 2.0.0
    forcestring: true
```
... but what's actually committed to git is the following and it deletes the original user-defined parameter overrides.
```yaml
helm:
  parameters:
  - name: foo.image.repository
    value: foo
    forcestring: true
  - name: foo.image.tag
    value: 2.0.0
    forcestring: true
```

And this image-updater behavior is a suspected cause of #562 and the likes
as image-updater ignores the tags for an image that it failed to fetch when
there are multiple updatable images defined in an `Application` and generate 
the parameter overrides without the ignored image. 
If only a part of image tags were successfully fetched from the image registry
due to unstable network, image-updater may write a broken parameter override
file to git.

### CODE REVIEW

File: pkg/argocd/update.go [hunk 1]
```diff
@@ -385,21 +386,46 @@ func marshalParamsOverride(app *v1alpha1.Application) ([]byte, error) {
...
 			Helm: helmParameters{
 				Parameters: app.Spec.Source.Helm.Parameters,
 			},
 		}
+
+		if len(originalData) == 0 {
+			override, err = yaml.Marshal(newParams)
+			break
+		}
+		err := yaml.Unmarshal(originalData, &params)
```
Comment (jannfis) [inline pkg/argocd/update.go:423]:
```suggestion
		err = yaml.Unmarshal(originalData, &params)
```

**Review by jannfis — CHANGES_REQUESTED** [review]
Nice one, thank you!

The linter complained, and it turns out your code shadows `err` in one place. Please fix it, otherwise LGTM :)

**Comment by jannfis** [comment]
Oh, and please make sure the DCO passes by signing off your commits

**Comment by ks-yim** [comment]
@jannfis
Thanks for taking your time to review this!

The issue described in the PR description has actually caused a disruption to our production(insignificant, though) workload so it would be very nice if this PR is shipped into the next release.

Resolved the variable shadowing issue and signed-off all my commits included in the PR.
I tried to make it pass `Spell checking` and `Run codegen` by merging the latest `master` but had no luck. 😅

**Comment by jannfis** [comment]
Thanks for looking into it, @ks-yim!

I have just merged #590 , so if you rebase your PR, the codegen step should pass.

I'm aiming to release v0.13.0 after https://github.com/argoproj-labs/argocd-image-updater/pull/578 has been merged, however, it needs some attention from the submitter.

**Review by jannfis — APPROVED** [review]
LGTM. Going to merge it without rebase required, because the codegen error is negligible.

**Pull request merged**